cmd/maya-apiserver/spc-watcher/handler.go

```diff
@@ -29,16 +29,25 @@ import (
 // syncHandler compares the actual state with the desired, and attempts to
 // converge the two. It then updates the Status block of the spcPoolUpdated resource
 // with the current status of the resource.
-func (c *Controller) syncHandler(key, operation string) error {
-	// getSpcResource will take a key as argument which conatins the namespace/name or simply name
+func (c *Controller) syncHandler(key, operation string, object interface{}) error {
```

Comment:
Passing both the key and the object may be duplicating things. But, seeing the need for handling the delete in case of no finalizers.